lib/file_store/base_store.rb

```diff
@@ -28,6 +28,10 @@ def remove_file(url, path)
       not_implemented
     end
 
+    def upload_path
```

Comment:
I've carefully refactored `upload_path` into `base_store.rb` so that we can use it on both `s3_store.rb` and `local_store.rb`. I had to add a couple of backslashes in `local_store.rb` for this to work. No side effects for this, according to the specs 😋